src/engraving/libmscore/measurerepeat.h

```diff
@@ -43,6 +43,8 @@ class MeasureRepeat final : public Rest
     MeasureRepeat* clone() const override { return new MeasureRepeat(*this); }
     EngravingItem* linkedClone() override { return EngravingItem::linkedClone(); }
 
+    static constexpr int MAX_NUM_MEASURES = 4;
+
     void setNumMeasures(int n) { m_numMeasures = n; }
```

Comment:
You should add IF_ASSERT_FAILED to this method to check if n <= MAX_NUM_MEASURES